rasa/core/actions/action.py

```diff
@@ -168,6 +175,59 @@ def __str__(self) -> Text:
         return "Action('{}')".format(self.name())
 
 
+class ActionUtterPredictedResponse(Action):
+    """An action which queries the Response Selector for the appropriate response."""
+
+    def __init__(self, name: Text, silent_fail: Optional[bool] = False):
+        self.action_name = name
+        self.silent_fail = silent_fail
+
+    async def run(
+        self,
+        output_channel: "OutputChannel",
+        nlg: "NaturalLanguageGenerator",
+        tracker: "DialogueStateTracker",
+        domain: "Domain",
+    ):
+        """Query the appropriate response and create a bot utterance with that."""
+
+        message = None
+
+        # Build query keys for tracker. This is a list because response for the action can be
+        # predicted by one of the two -
+        # 1. a specific response selector of this open domain intent type
+        # 2. a generic response selector built for all open domain intent types in the training data.
+        # We check in both
+        response_selector_keys = [
+            "{0}{1}".format(self.action_name, OPEN_UTTERANCE_KEY_SUFFIX),
+            DEFAULT_OPEN_UTTERANCE_TYPE_KEY,
+        ]
+        for query_key in response_selector_keys:
+            logger.debug("Looking for response with query_key {}".format(query_key))
+            if query_key in tracker.latest_message.parse_data:
+                logger.debug("Picking response of type {}".format(query_key))
+                message = {
+                    "text": tracker.latest_message.parse_data.get(query_key).get("name")
```

Comment:
since you checked that it is in the dict, use a direct access instead
```suggestion
                    "text": tracker.latest_message.parse_data[query_key].get("name")
```